patterns/Prototype/README.md

```diff
@@ -0,0 +1,112 @@
+# Prototype Design Pattern
+
+
+The Prototype Design Pattern is a creational pattern that enables the creation of new objects by copying an existing object. Prototype allows us to hide the complexity of making new instances from the client. The concept is to copy an existing object rather than create a new instance from scratch, something that may include costly operations. The existing object acts as a prototype and contains the state of the object.
+<br>
+## Prototype Pattern in JavaScript
+
+It promotes the creation of objects without specifying the exact class of object to create. This pattern is particularly useful when you want to create instances of complex objects efficiently.
+<br>
```

Comment:
What is particularly interesting about this example is the fact that JavaScript is a prototypal language. This means that all JavaScript objects _have_ a prototype already (you can see it by using `.prototype` on any JS object), and so the prototype design pattern is supported in JavaScript out of the box.

In other words, you don't really need to do anything in JS to implement the prototype pattern - it's already there via `Object.create()` and `Object.assign()`! Contrast the JavaScript implementation to the Java implementation, for example - the Java implementation is necessarily much more complex, because Java does not have a prototype object.

However, I think this means you have another responsibility here: You should state that as a prototypal language, JS supports prototype pattern without any real additional code needed. See if you can find some sources to help you state this clearly here.